### DISCUSSION TITLE
where am i doing wrong

### DISCUSSION THREAD

**Discussion by firatoltulu** [linq2db/linq2db#4091] [General]
i get error  "Npgsql.PostgresException (0x80004005): 42703: column "year" does not exist."

**Linq query,**
```cs
 var query = filter.GroupBy(x => new
                {
                    Year = Sql.DatePart(Sql.DateParts.Year, x.CreatedOnUtc),
                    Month = Sql.DatePart(Sql.DateParts.Month, x.CreatedOnUtc),
                    Day = Sql.DatePart(Sql.DateParts.Day, x.CreatedOnUtc)
                })
                .Select(g => new { Period = new { Year = g.Key.Year, Month = g.Key.Month, Day = g.Key.Day }, Count = g.Count() });
```
**here to output.**
```sql
SELECT
	t1."Key_1",
	t1."Key_2",
	t1."Key_3",
	Count(*)
FROM
	(
		SELECT
			DatePart(year, "selectParam"."CreatedOnUtc") as "Key_1",
			DatePart(month, "selectParam"."CreatedOnUtc") as "Key_2",
			DatePart(day, "selectParam"."CreatedOnUtc") as "Key_3"
		FROM
			"Accounts" "selectParam"
				INNER JOIN "Account_RoleMappings" r ON "selectParam"."Id" = r."AccountId"
	 
	) t1
GROUP BY
	t1."Key_1",
	t1."Key_2",
	t1."Key_3"
```

**Comment by sdanyliv** [comment]
Which type has `CreatedOnUtc`? DateTimeOffset?

**Reply by MaceWindu** [reply]
It doesn't make much sense.

Query sent to PostgreSQL, but we don't generate `DatePart` function for it  (and postgresql doesn't have such function).

**Reply by MaceWindu** [reply]
SOunds like context misconfiguration, but I cannot guess how it is possible

**Reply by firatoltulu** [reply]
> Which type has `CreatedOnUtc`? DateTimeOffset?

         /// <summary>
        /// Gets or sets the date and time of entity creation
        /// </summary>
        public DateTime CreatedOnUtc { get; set; }

**Reply by firatoltulu** [reply]
> SOunds like context misconfiguration, but I cannot guess how it is possible

How can i do grouping year and month with linq2db ?

**Reply by MaceWindu** [reply]
You do it correctly, but generated SQL looks like SQL for SQL Server. Check/show your connection configuration - for some reason it use SQL Server query generation instead of PostgreSQL.

**Reply by firatoltulu** [reply]
you can here to connection detail

 
![resim](https://user-images.githubusercontent.com/4473771/234701153-08d2bc43-6571-486f-bed6-24c86132a5a8.png)

![resim](https://user-images.githubusercontent.com/4473771/234700985-927f907c-e437-4486-ba30-bdbdf76ba03f.png)

**Reply by sdanyliv** [reply]
Do you have any manipulations with `MappingSchema.Default`? Or something with `MappingSchema`?

**Reply by MaceWindu** [reply]
I see you use `DataContext`. We recently discovered issue with data context (will be fixed next release), which could result in similar issue. Could you try to swith to `DataConnection` and report back results?

**Reply by firatoltulu** [reply]
I switch to dataconnection but i get many errors. one of them maxpooling.  can create issue for this?

**Reply by MaceWindu** [reply]
Create new issue and provide isolated runnable example

**Reply by firatoltulu** [reply]
> Do you have any manipulations with `MappingSchema.Default`? Or something with `MappingSchema`?

Yes i did, customize mapping schema. 

 var mappingSchema = new MappingSchema(ConfigurationName)
                {
                    MetadataReader = new FluentMigratorMetadataReader(this)
                };

**Reply by MaceWindu** [reply]
I suspect by `FluentMigratorMetadataReader` you mean `Nop.Data.Mapping.FluentMigratorMetadataReader` from NopCommerce. It uses linq2db v4, so my comment about `DataContext` probably is not valid.

What is `ConfigurationName`?

**Reply by firatoltulu** [reply]
PostgreSQLVersion.v15, i change datacontext switch to data connection. it's fixed.

**Reply by MaceWindu** [reply]
> PostgreSQLVersion.v15

could be a reason. Not a good idea to use linq2db provider names as configuration name. If you need to define named configuration it should be something neutral like "CustomersDatabase"
